public/app/features/datasources/components/picker/DataSourceDropdown.test.tsx

```diff
@@ -0,0 +1,254 @@
+import { findByText, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { UserEvent } from '@testing-library/user-event/dist/types/setup/setup';
+import React from 'react';
+
+import { DataSourceInstanceSettings, DataSourcePluginMeta, PluginMetaInfo, PluginType } from '@grafana/data';
+import { ModalRoot, ModalsProvider } from '@grafana/ui';
+import config from 'app/core/config';
+import { defaultFileUploadQuery } from 'app/plugins/datasource/grafana/types';
+
+import { DataSourceDropdown } from './DataSourceDropdown';
+import * as utils from './utils';
+
+const pluginMetaInfo: PluginMetaInfo = {
+  author: { name: '' },
+  description: '',
+  screenshots: [],
+  version: '',
+  updated: '',
+  links: [],
+  logos: { small: '', large: '' },
+};
+
+function createPluginMeta(name: string, builtIn: boolean): DataSourcePluginMeta {
+  return { builtIn, name, id: name, type: PluginType.datasource, baseUrl: '', info: pluginMetaInfo, module: '' };
+}
+
+function createDS(name: string, id: number, builtIn: boolean): DataSourceInstanceSettings {
+  return {
+    name: name,
+    uid: name + 'uid',
+    meta: createPluginMeta(name, builtIn),
+    id,
+    access: 'direct',
+    jsonData: {},
+    type: '',
+    readOnly: true,
+  };
+}
+
+const mockDS = createDS('mockDS', 1, false);
+const xMockDS = createDS('xMockDS', 2, false);
```

Comment:
```suggestion
const mockDS1 = createDS('mock.datasource.1', 1, false);
const mockDS2 = createDS('mock.datasource.2', 2, false);
```
Totally opinionated preference, but I find it easier to read when calling the mocked data as simple as possible 😄 xMockDS seems to be something with logic itself.